src/condition/cooldown.rs

```diff
@@ -0,0 +1,233 @@
+use core::time::Duration;
+use bevy::prelude::*;
+
+use super::DEFAULT_ACTUATION;
+use crate::prelude::*;
+
+/// Returns [`ActionState::Fired`] when input is being pressed after a set duration passed or for the first time.
+///
+/// Returns [`ActionState::None`] when the action timer is not finished or already actuated last frame.
+#[derive(Component, Reflect, Debug, Clone)]
+pub struct Cooldown {
+    /// Trigger threshold.
+    pub actuation: f32,
+    
+    actuated: bool,
+
+    /// The type of time used to advance the timer.
+    pub time_kind: TimeKind,
```

Comment:
Could you swap places these 2? So public fields come first :slightly_smiling_face: